middleware/validateParamsIDwithToken.js

```diff
@@ -0,0 +1,19 @@
+require("express-async-errors");
+const jwt = require("jsonwebtoken");
+
+module.exports = (paramsNameToCompare) => {
```

Comment:
Please be cautious to apply the comments above here is as well where it makes sense.